File: nomad-samples/common/js/account-authenticaton/account/forgot-password.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function forgotPassword(USERNAME) 
{
		// Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${sessionStorage.getItem("token")}`);
  
  	const BODY = {
        username: USERNAME
    };
  
  // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/forgot-password`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        return(RESPONSE.status);
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Refresh Token failed");
}

File: nomad-samples/common/py/account_authentication/account/forgot_password.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def forgot_password(USENAME: str, AUTH_TOKEN: str) -> None:
    if not USENAME:
        raise Exception("USENAME: The username is invalid")

    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")
        
    API_URL = PORTAL_API_URL + "/account/forgot-password"
  
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json"
    }

    # Build the payload body
    BODY = {
        "username": USENAME
    }
    
    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        
        if not RESPONSE.ok:
            raise Exception()
          
    except:
        api_exception_handler(RESPONSE, "Forgot Password Failed")